src/olympia/amo/tests/test_send_mail.py

```diff
@@ -280,10 +281,12 @@ def test_async_will_retry(self, backend):
             send_mail('test subject',
                       'test body',
                       recipient_list=['somebody@mozilla.org'])
-        send_mail('test subject',
-                  'test body',
-                  async=True,
-                  recipient_list=['somebody@mozilla.org'])
+
+        with self.assertRaises(Retry):
+            send_mail('test subject',
+                      'test body',
+                      async=True,
+                      recipient_list=['somebody@mozilla.org'])
```

Comment:
Not sure I understand this replacement - so now if there's a `RuntimeError` on the first try it fails on the second try?  It never succeeds?

Reply:
It does but now it raises a `Retry` exception that get's handled by the worker and retried properly. This didn't happen before for various things that masked that behavior and actually retried running the task during the tests. But we were never able to see that, now we are.

Reply:
afaik this changed with https://github.com/celery/celery/pull/2607/commits/7bf736bfae86766819c3af5da661f7152b364c85